backend/feed_service.py

```
@@ -69,6 +72,58 @@ def parse_published_time(entry):
+
+        for res in addr_info:
+            ip_str = res[4][0]
+            try:
+                ip = ipaddress.ip_address(ip_str)
+                if ip.is_private or ip.is_loopback or ip.is_link_local or ip.is_reserved:
+                    logger.warning(f"Blocked SSRF attempt: {url} resolved to restricted IP {ip}")
+                    return False
+            except ValueError:
+                continue
+
+        return True
```

Ruff (0.14.10)

Consider moving this statement to an `else` block

(TRY300)